runtime/onert/backend/train/ops/PoolLayer.cc

```diff
@@ -35,14 +147,32 @@ void PoolLayer::configure(const IPortableTensor *input, const uint32_t paddingLe
                           const uint32_t paddingBottom, const uint32_t strideWidth,
                           const uint32_t strideHeight, const uint32_t kernelWidth,
                           const uint32_t kernelHeight, const ir::Activation activation,
-                          IPortableTensor *output, const PoolType op_type)
+                          IPortableTensor *output, const PoolType op_type,
+                          IPortableTensor *deriv_input, const IPortableTensor *deriv_output)
 {
+  _input = input;
+  _output = output;
+
+  _deriv_output = deriv_output;
+  _deriv_input = deriv_input;
+
+  // ready inference kernel
+  cpu::ops::PoolLayer::configure(input, paddingLeft, paddingRight, paddingTop, paddingBottom,
+                                 strideWidth, strideHeight, kernelWidth, kernelHeight, activation,
+                                 output, convert_to_infer(op_type));
+
+  if (output->data_type() != OperandType::FLOAT32)
+  {
+    throw std::runtime_error("PoolLayer : Unsupported data type for training");
+  }
+
+  // ready training kernel
```

Comment:
Both Inference and training kernel is ready in compile time.
In Runtime, depends on `bool training` value, It decides which kernel to use.